contracts/cw20-stake-external-rewards/src/contract.rs

```diff
@@ -99,14 +100,41 @@ pub fn instantiate(
         )
         .add_attribute("reward_rate", reward_config.reward_rate)
         .add_attribute("period_finish", reward_config.period_finish.to_string())
-        .add_attribute("reward_duration", reward_config.reward_duration.to_string()))
+        .add_attribute("reward_duration", reward_config.reward_duration.to_string())
+        .add_attribute("is_paused", "false"))
 }
 
 #[cfg_attr(not(feature = "library"), entry_point)]
-pub fn migrate(deps: DepsMut, _env: Env, _msg: MigrateMsg) -> Result<Response, ContractError> {
-    // Set contract to version to latest
+pub fn migrate(deps: DepsMut, _env: Env, msg: MigrateMsg) -> Result<Response, ContractError> {
     set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
-    Ok(Response::default())
+    match msg {
+        MigrateMsg::FromV1 { is_paused } => {
+            // Load the current config and introduce the claim pause flag
+            let current_config =
+                stake_cw20_external_rewards_v1::state::CONFIG.load(deps.storage)?;
+            let is_paused = if let Some(val) = is_paused {
+                val
+            } else {
+                false
+            };
+
+            let new_config = Config {
+                owner: current_config.owner,
+                manager: current_config.manager,
+                staking_contract: current_config.staking_contract,
+                reward_token: current_config.reward_token,
+                is_paused,
+            };
+            deps.storage.set(b"config", &to_vec(&new_config)?);
```

Comment:
in general we want to try to avoid overriding previous state just to be safe. we can instead create a `config_v2` state key and write to that. take look at the migrate method in `cw-proposal-single` to see an example of that :)